common-npm-packages/docker-common-v2/containerimageutils.ts

```diff
@@ -72,4 +73,125 @@ export function getResourceName(image: string, digest: string) {
     namespace = namespace ? namespace + '/' : '';
     
     return "https://" + registry  + namespace  + repository + "@sha256:" + digest;
-  }
\ No newline at end of file
+  }
+
+export function getFinalBaseImageName(dockerFilePath: string): string {
```

Comment:
Continuing the discussion from:
https://github.com/microsoft/azure-pipelines-tasks/pull/14695#discussion_r626806057

Had a chat with @ds-ms. It makes sense to replace the existing `getBaseImageName` method as this is enhancing the logic.

Reply:
Is there a way that we can test that we are not breaking other people scenarios. The expectations is that this new logic  should enhance the previous one but I would feel more confident if I see some kind of testing. I couldn't find a tests for the common packages functions.

Reply:
Good point. Having L0 tests would definitely have helped here. Unfortunately, its not there so we will need to rely on our own assessment.
It seems safe in the sense - It is guaranteed to give the result that the previous logic was giving or better.